receiver/httpcheckreceiver/scraper.go

```diff
@@ -32,6 +34,46 @@ type httpcheckScraper struct {
 	mb       *metadata.MetricsBuilder
 }
 
+// Custom transport to measure DNS resolution time
+func newCustomTransport(insecureSkipVerify bool, dnsDuration *time.Duration) *http.Transport {
+	return &http.Transport{
```

Comment:
please find a way to use the confighttp lib for this and inject what you need. If you need to make changes to confighttp to support this, so be it.